internal/mesh/internal/controllers/meshgateways/controller.go

```diff
@@ -24,7 +28,47 @@ func Controller() *controller.Controller {
 
 type reconciler struct{}
 
+// Reconcile is responsible for creating a Service w/ a MeshGateway owner,
+// in addition to other things discussed in the RFC.
 func (r *reconciler) Reconcile(ctx context.Context, rt controller.Runtime, req controller.Request) error {
+	rt.Logger = rt.Logger.With("resource-id", req.ID)
+	rt.Logger.Trace("reconciling mesh gateway")
+
+	service := &pbcatalog.Service{
+		Workloads: &pbcatalog.WorkloadSelector{
+			// Assumes that our MeshGateway Deployment in K8s is named the same as the MeshGateway resource in Consul
+			// TODO(nathancoleman) Fetch the MeshGateway and use WorkloadSelector from there
+			Prefixes: []string{req.ID.Name},
+		},
+		Ports: []*pbcatalog.ServicePort{
+			{
+				Protocol:    pbcatalog.Protocol_PROTOCOL_TCP,
+				TargetPort:  "wan",
+				VirtualPort: 8443,
+			},
+			{
+				Protocol:   pbcatalog.Protocol_PROTOCOL_MESH,
+				TargetPort: "mesh",
```

Comment:
Can we make these hardcoded things into constants? Or is there already a constant for the `mesh` name of this port?

Reply:
I can make constants for them. In the longer term, we need to add this `WorkloadSelector` onto the `MeshGateway` itself instead of assuming values based on what I happen to know consul-k8s is doing atm

Reply:
Added constants and a TODO referencing the future work to add ports + selector onto the `MeshGateway` resource